### ISSUE TITLE
Changing labels of classes are not reflected correctly in the tree

### ISSUE THREAD

**Issue by csnyulas** [protegeproject/webprotege#433]
If the classes are using rdfs:label as their display label, changing it, will not be reflected in the class tree. In fact, the old label of the class is continued to be displayed, but all of its children will disappear. **Refreshing the tree would not help either.**

Before:
![image](https://user-images.githubusercontent.com/2441046/33855568-247840c2-de7a-11e7-8ba0-2e3174e978a7.png)

After changing rdfs:label and refreshing the tree:
![image](https://user-images.githubusercontent.com/2441046/33855615-4e244ef2-de7a-11e7-9fb2-54c6b0870fcc.png)

**Comment by csnyulas** [comment]
There are no errors shown in the logs, but there were some warnings about low memory, like this (I forgot to assign appropriate heap size to tomcat):
```
Low Memory: Using 53 MB of 123 MB (43.19%) [70 MB free]
```

I am not sure if this had anything to do with the refresh problem or not, since I did many different operations (most notably a lot of class hierarchy changes, especially with drag&drop), but after  restarting tomcat with more memory, I can't reproduce the behavior at the moment either. I will play with it some more, before closing the issue.

**Comment by csnyulas** [comment]
OK, I managed to reproduce the bug...

It happens on newly created entities. If you create a class, for example, and you change its `rdfs:label` the class tree will not pick up the changes, even after a refresh. If you create a subclass for it, after refreshing the tree again, you won't even be able to see this subclass, since the class shown in the tree (with the old name) does not exist in the ontology. This issue does not affect the changing of the label of the "old classes", which seem to work properly, both before and after encountering this particular problem.

I recorded [this video](https://stanford.box.com/s/w8oo85kqma2p2ojjgknssaxh5bra0ptg) to demonstrates the problem.

Other class trees (in different tabs) seem to be refreshed as expected. For this class tree however, only a browser refresh would help.
